xml/hwbp_registers.xml

```diff
@@ -376,6 +376,10 @@
             trigger will be taken if and only if all the triggers in the chain
             match at the same time.
 
+            It is allowed to terminate a chain with a trigger with
+            $|type| \neq 2$. In most situations this results in a
+            chain that can never match.
+
```

Comment:
I worry that this wording isn't precise enough.  Is what you are really saying is that if there is a trigger chain that includes a trigger with a type other than 2, it is allowed for an implementation to never match the chained trigger?

Reply:
I phrased this to (hopefully) make hardware implementations simple so they don't have to look at the trigger type of the next trigger. Debuggers can set the chain bit if they want, and if it turns out to make sense, so much the better. But I don't want the hardware to have to do a bunch of extra checks in case a debugger is trying to be clever.

If you chain type 2/6 to:
* icount (3), unclear what exactly happens with the new pending bit, because we don't define a "match" for icount.
* itrigger (4), we don't define match either. I think typically the hart either takes an interrupt or executes an instruction, so probably never matches.
* etrigger (5), the same as for itrigger. You never really do a load/store *and* take an exception. A memory exception means the load/store didn't happen.
* tmexttrigger (7), don't define match, I assume this is something that's evaluated in between instruction executions, so again can't match together.

Maybe I should have written:
Debuggers should not terminate a chain with a trigger with a different type. It is implementation-specific when exactly such a chain fires.

Reply:
Or... a trigger that chains more than one trigger type has undefined behavior.

Reply:
Yes, but we don't actually have a definition of what we mean by "undefined." It's fine for these triggers to fire at strange times, but it's not OK for such a setup to lock up the trigger unit.